staticaddr/withdraw/manager.go

```diff
@@ -380,7 +380,39 @@ func (m *Manager) createFinalizedWithdrawalTx(ctx context.Context,
 		).FeePerKWeight()
 	}
 
+	// We'll now check the selected fee rate leaves a withdrawal output that
```

Comment:
I propose to cover the logic and various sub-cases in this method with unit tests. Maybe factor out a pure function to make it unit tests friendly.